src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -862,6 +863,14 @@ else if (!getGuild().getSelfMember()
     @Override
     public AuditableRestAction<Void> suppressEmbeds(boolean suppressed)
     {
+        if (!getJDA().getSelfUser().equals(getAuthor()))
+        {
+            TextChannel textChannel = getTextChannel();
+            if (isFromType(ChannelType.PRIVATE))
```

Comment:
It looks like you call `getTextChannel()` before checking the type, this means it will throw before this check ever happens.